File: problem_015.py
#!/usr/bin/env python

print("Project Euler - Problem 15:")
print("Starting in the top left corner in a 20 by 20 grid, how many routes are there to the bottom right corner?\n")

triangle = [[1,0],[1,1,0]]

# Build pascal's triangle
for r in range(2, 41):
    triangle.append([1])
    for n in range (1, r+1):
        triangle[r].append(triangle[r-1][n] + triangle[r-1][n-1])
    triangle[r].append(0)

# Print middle value of last line
print(triangle[40][20])

File: problem_015.js
#!/usr/bin/node

console.log("Project Euler - Problem 15:");
console.log("Starting in the top left corner in a 20 by 20 grid, how many routes are there to the bottom right corner?\n");

const limit = 20;
var triangle = [[1, 0]];

for (var i=0; i<limit * 2; i++){
    let new_row = [1];
    for (var n=0; n<triangle[i].length-1; n++)
	new_row.push(triangle[i][n] + triangle[i][n+1]);
    new_row.push(0);
    triangle.push(new_row);
}

console.log(triangle[40][20]);
